### PR TITLE
Add a security response header filter

### PR DESCRIPTION
**Pull request by gwicke**
Move REST API specific security response header handling from hyperswitch to
restbase.

### CODE REVIEW

**Comment by gwicke** [comment]
See https://github.com/wikimedia/hyperswitch/pull/60 for the corresponding hyperswitch change.

File: test/features/parsoid/ondemand/ondemand.js [hunk 1]
```diff
@@ -222,6 +222,7 @@ describe('on-demand generation of html and data-parsoid', function() {
             uri: pageUrl + '/html/' + title
         })
         .then(function (res) {
+            console.log(res);
```
Comment (Pchelolo) [inline test/features/parsoid/ondemand/ondemand.js]: oh

File: lib/security_response_header_filter.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
+"use strict";
+const P = require('bluebird');
```
Comment (Pchelolo) [inline lib/security_response_header_filter.js]: We don't use this yet in RB

**Comment by Pchelolo** [comment]
LGTM, though needs a bump in `hyperswitch` dependency version.

File: lib/security_response_header_filter.js (hunk 2 above)
Comment (gwicke) [inline lib/security_response_header_filter.js]: {{done}}

**Comment by Pchelolo** [comment]
And travis is still unhappy

**Comment by gwicke** [comment]
Corresponding puppet patch at https://gerrit.wikimedia.org/r/#/c/309486/.

**Comment by Pchelolo** [comment]
@gwicke travis yet again..

**Comment by d00rman** [comment]
FWIW, I think it would have been better to leave the capacity in HyperSwitch, but remove only the RB-specific parts and transfer them here.

**Comment by gwicke** [comment]
@d00rman: The policies on security headers are all quite specific to the API / use case, which is why I moved this to RESTBase. For example, the ServiceWorker proxy needs to allow inline scripts, while the content API is keen on disallowing those.

**Comment by d00rman** [comment]
> The policies on security headers are all quite specific to the API / use case, which is why I moved this to RESTBase. For example, the ServiceWorker proxy needs to allow inline scripts, while the content API is keen on disallowing those.

Sure, sure. I meant that IMHO HyperSwitch should have kept an `addCSP` or similar method, when clients would just then give it the values, and for any value not given HyperSwitch would have a sane default.

**Comment by gwicke** [comment]
@d00rman: If the filter proves useful for other use cases (which seems not that unlikely), then we could move it to a separate package, or even start a collection of bundled filters in hyperswitch itself. Right now, all API-specific filters live in RB, which is why I moved this one over.

**Pull request merged**